### PR TITLE
paperless-ngx: 1.11.3 -> 1.12.2

### PR DESCRIPTION
**Pull request by mweinelt**
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.12.0
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.12.1
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.12.2

Ok, rundown of what I did.

- The frontend is now available in a separate derivation through `paperless-ngx.frontend`
- Started to selectively copy content into `$out`, specifically left out
  - `docs` (probably needs to be built using mkdocs, but then into a dedicated output)
  - `docker`
  - `scripts` (systemd units and docker startup script)
- Dropped the tests and its samples from the final package to save 14 MB, we're now at ~90.4 MB down from 187 MB
  - 82 MB Frontend
  - 8.4 MB Backend
- Updated to 1.12.2
- Added a migration, so we're reindexing documents as recommended in the 1.12.1 release notes

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/paperless.nix [hunk 1]
```diff
@@ -230,6 +230,13 @@ in
           ${pkg}/bin/paperless-ngx migrate
           echo ${pkg} > "$versionFile"
         fi
+
+        # Reindex once after 1.12.x upgrade
+        if [[ -f "${cfg.dataDir}/.1_12_document_reindex" ]]: then
```
Comment (erikarvstedt) [inline nixos/modules/services/misc/paperless.nix:235]: Should be `if [[ ! -f ...`.

File: nixos/modules/services/misc/paperless.nix [hunk 2]
```diff
@@ -230,6 +230,13 @@ in
...
+          echo "Reindixing documents once, required after the 1.12.x upgrade"
```
Comment (erikarvstedt) [inline nixos/modules/services/misc/paperless.nix:236]: Reind**e**xing

**Comment by mweinelt** [comment]
Ok, rundown of what I did.

- The frontend is now available in a separate derivation through `paperless-ngx.frontend`
- Started to selectively copy content into `$out`, specifically left out
  - `docs` (probably needs to be built using mkdocs, but then into a dedicated output)
  - `docker
  - `scripts` (systemd units and docker startup script)
- Dropped the tests and its samples from the final package to save 14 MB, we're now at ~41.5 MB
  - 34 MB Frontend
  - 7.5 MB Backend

File: pkgs/applications/office/paperless-ngx/default.nix [hunk 3]
```diff
@@ -255,10 +266,11 @@ python.pkgs.buildPythonApplication rec {
       --prefix PATH : "${path}"
   '';
 
-  nativeCheckInputs = with python.pkgs.pythonPackages; [
+  nativeCheckInputs = with python.pkgs; [
+    factory_boy
```
Comment (erikarvstedt) [inline pkgs/applications/office/paperless-ngx/default.nix:279]: `factory_boy` is already in this list.

Comment (mweinelt) [reply pkgs/applications/office/paperless-ngx/default.nix:279]: Fixup pushed.

**Review by erikarvstedt — APPROVED** [review]
- [x] `nixosTests.paperless` succeeds on `x86_64-linux`
- [x] I've manually tested the frontend in a container-based deployment, adding a few docs and checking their consumption

**Comment by gador** [comment]
- [x] nixpkgs-review shows no errors
- [x] `nix-build -A nixosTests.paperless` succeeds 
- [x] running paperless within a Gnome VM works just fine. Adding and saving documents as well

**Review by gador — APPROVED** [review]

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
Edit: Ready to merge.

**Review by erikarvstedt — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Pull request merged**